packages/fileeditor/src/toc/markdown.ts

```diff
@@ -18,6 +20,16 @@ export class MarkdownTableOfContentsModel extends TableOfContentsModel<
   IEditorHeading,
   IDocumentWidget<FileEditor, DocumentRegistry.IModel>
 > {
+  protected parser: IMarkdownParser | null;
+
+  constructor(
+    widget: IDocumentWidget<FileEditor>,
+    parser: IMarkdownParser | null,
+    configuration?: TableOfContents.IConfig | undefined
```

Comment:
Can you move `parser` to the end of the arguments list so that we don't break compatibility with downstream code which used `new MarkdownTableOfContentsModel(widget, configuration)`?